Enable UseShellExecute on all platforms

SteveL-MSFT opened the pull request:
## PR Summary

Previously, we limited use of UseShellExecute when starting a new process to only Windows Desktop.  However, corefx now supports UseShellExecute (appropriately calling `open` on macOS and `xdg-open` on Linux) so we can have consistent code and remove some of the existing redundant code or unnecessary limiting code.

There is a [bug](https://github.com/dotnet/corefx/issues/30718) in corefx where Arguments is passed to the executable used to `ShellExecute` such as `open` on macOS instead of the FileName intended to be started, so don't use `ShellExecute` if arguments is passed.

Also, because of difference in behavior with using `ShellExecute` with console apps compared to Windows where a new console window is opened, we only use `ShellExecute` if `FileName` doesn't refer to an actual command, like a URL.

Fix https://github.com/PowerShell/PowerShell/issues/5715

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [x] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

SteveL-MSFT committed:
corefx now supports UseShellExecute on non-Windows, removing unncessary code

SteveL-MSFT committed:
address mklement0's feedback

iSazonov commented on test/powershell/Modules/Microsoft.PowerShell.Utility/Invoke-Item.Tests.ps1:
```diff
@@ -219,4 +189,26 @@ Describe "Invoke-Item tests on Windows" -Tags "CI","RequireAdminOnWindows" {
         Test-Path $renamedtestfilepath | Should -BeTrue
     }
+
+    It "Should invoke an executable file without error" -Skip:(!$IsWindows) {
```
Please clarify why we remove previous all-platform test and add this new for Windows only?

SteveL-MSFT replied:
The previous test expected UseShellExecute to only be available on Windows and the previous code was checking for different behavior on non-Windows.  Since I've changed it to only use ShellExecute on non-Windows if the target is not an executable, I should put back this test.

SteveL-MSFT committed:
[feature]
put back Invoke-Item test after recent code change

iSazonov commented on src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs:
```diff
@@ -1835,6 +1834,14 @@ protected override void BeginProcessing()
+                //so don't use ShellExecute if arguments are specified
+
+                //Linux relies on `xdg-open` and macOS relies on `open` which behave differently than Windows ShellExecute when running console commands
+                //as a new console will be opened.  So to avoid that, we only use ShellExecute on non-Windows if the filename is not an actual command (like a URI)
```
Please add space after `//` in comments.

SteveL-MSFT replied:
I was following the existing comments

iSazonov replied:
Yes, I see. But if we continue use CodeFactor we have to fix every code block without following file style.

SteveL-MSFT replied:
I'll update this file with whitespace as a separate commit so it doesn't make the overall code review noisy

iSazonov reviewed (approved):
LGTM with minor comment.

SteveL-MSFT committed:
add whitespace to comments, removed some obvious not useful comments

iSazonov commented on src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs:
```diff
@@ -428,7 +428,7 @@ internal static bool TryHasExited(Process process)
 
         #endregion Internal
-    }//ProcessBaseCommand
+    } // ProcessBaseCommand
```
I would delete such comments. They are useless in modern IDEs.

iSazonov commented on src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs:
```diff
@@ -520,7 +520,7 @@ public SwitchParameter IncludeUserName
 
         ///<summary>
-        ///To display the modules of a process
+        /// To display the modules of a process
```
Please add final dot.

iSazonov commented on src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs:
```diff
@@ -530,7 +530,7 @@ public SwitchParameter IncludeUserName
 
         ///<summary>
-        ///To display the fileversioninfo of the main module of a process
+        /// To display the fileversioninfo of the main module of a process
```
Please add final dot.

SteveL-MSFT committed:
quick pass through comments removing unnecessary ones, capitalizing and adding period to end of summaries

SteveL-MSFT commented:
@iSazonov did a pass through the comments, removed a bunch of unnecessary ones and added punctuation to the summaries.  However, this PR has grown much larger than the original issue.  I think in general, we want CodeFactor issues outside of the changes to be a separate PR.

iSazonov commented:
@SteveL-MSFT I agree that large style fixes we should place in separate PR. A separate style fix commit is suitable for small changes.
